scanner/src/main/kotlin/experimental/ExperimentalScanner.kt

```diff
@@ -178,6 +171,22 @@ class ExperimentalScanner(
         return controller.getNestedScanResultsByPackage()
     }
 
+    private fun resolvePackageProvenances(controller: ScanController) {
+        log.info { "Resolving provenance for ${controller.packages.size} packages." }
+
+        val mark = TimeSource.Monotonic.markNow()
```

Comment:
Why not `measureTime`?

Reply:
To reduce nesting. I find `measureTime` is often very invasive if the block is longer than 1-2 lines because it looks like control flow but is just measuring a duration.

Reply:
I see. On the upside, `measureTime` nicely groups *what*'s being measured, which is esp. useful for more than 1-2 lines, IMO. I'm not entirely happy about the use of something else than `measureTime` here (mostly for consistency reasons), but I also won't insist on changing it.

Reply:
Alternatively, the caller could now measure the time for the single function call.